blob/service.go

```diff
@@ -66,15 +80,21 @@ func DefaultSubmitOptions() *SubmitOptions {
 	}
 }
 
-// Submit sends PFB transaction and reports the height in which it was included.
+// Submit sends PFB transaction and reports the height at which it was included.
 // Allows sending multiple Blobs atomically synchronously.
 // Uses default wallet registered on the Node.
 // Handles gas estimation and fee calculation.
-func (s *Service) Submit(ctx context.Context, blobs []*Blob, options *SubmitOptions) (uint64, error) {
+func (s *Service) Submit(ctx context.Context, blobs []*Blob, gasPrice GasPrice) (uint64, error) {
 	log.Debugw("submitting blobs", "amount", len(blobs))
 
-	if options == nil {
-		options = DefaultSubmitOptions()
+	options := DefaultSubmitOptions()
+	if gasPrice > 0 {
+		blobSizes := make([]uint32, len(blobs))
+		for i, blob := range blobs {
+			blobSizes[i] = uint32(len(blob.Data))
+		}
+		options.GasLimit = blobtypes.EstimateGas(blobSizes, appconsts.DefaultGasPerBlobByte, auth.DefaultTxSizeCostPerByte)
+		options.Fee = types.NewInt(int64(math.Ceil(float64(gasPrice) * float64(options.GasLimit)))).Int64()
 	}
```

Comment:
Should gasPrice equal to 0 also be handled here? Otherwise, 0 falls under the default (-1) case.

Reply:
Nice catch, yes. 0 is a valid gas price

Reply:
Very good catch, thank you